File: traininglog-client/src/types/workout.ts
// Mirrors the C# enum WorkoutType exactly — values must match (Run = 0, CrossTrain = 1, Rest = 2)
export const WorkoutType = {
  Run: 0,
  CrossTrain: 1,
  Rest: 2,
} as const;

export type WorkoutType = (typeof WorkoutType)[keyof typeof WorkoutType];

export interface WorkoutDto {
  id: number;
  date: string;
  type: WorkoutType;
  plannedDistanceMiles: number | null;
  actualDistanceMiles: number | null;
  plannedPace: string | null;
  actualPace: string | null;
  duration: string | null;
  notes: string | null;
  completed: boolean;
  trainingPlanId: number;
  shoeId: number | null;
}

export interface CreateWorkoutDto {
  date: string;
  type: WorkoutType;
  plannedDistanceMiles: number | null;
  actualDistanceMiles: number | null;
  plannedPace: string | null;
  actualPace: string | null;
  duration: string | null;
  notes: string | null;
  completed: boolean;
  trainingPlanId: number;
  shoeId: number | null;
}

File: TrainingLog-api/TrainingLog.Domain/Entities/Workout.cs
namespace TrainingLog.Domain.Entities
{
    public enum WorkoutType
    {
        Run,
        CrossTrain,
        Rest
    }

    public class Workout
    {
        public int Id { get; set; }
        public DateTime Date { get; set; }
        public WorkoutType Type { get; set; }
        public decimal? PlannedDistanceMiles { get; set; }
        public decimal? ActualDistanceMiles { get; set; }
        public TimeSpan? PlannedPace { get; set; }
        public TimeSpan? ActualPace { get; set; }
        public TimeSpan? Duration { get; set; }
        public string? Notes { get; set; }
        public bool Completed { get; set; } = false;

        // Foreign keys
        public int TrainingPlanId { get; set; }
        public TrainingPlan? TrainingPlan { get; set; }

        public int? ShoeId { get; set; }
        public Shoe? Shoe { get; set; }
    }
}
